AWSAuthSDK/Tests/AWSMobileClientTests/AWSMobileClientTests.swift

```diff
@@ -241,6 +241,38 @@ class AWSMobileClientTests: XCTestCase {
         signIn(username: username)
     }
     
+    func testSignInWithListener() {
```

Comment:
I think part of this test (based on the `enforceOrder` in the `wait`, is to assert the order of callbacks, so make that part of the test name. Alternately, I've been thinking about using a pattern I've seen elsewhere with having tests prepended with a `given/when/then` comment block:

```swift
/// Tests successful sign in and order of callbacks
///
/// - Given: An unauthenticated session
/// - When:
///    - I add a user state listener
///    - I invoke `signIn` with a completion callback
/// - Then:
///    - The user state is `signedIn`
///    - My user state listener callback is invoked first
///    - My `signIn` completion callback is invoked last
```

What do you think? Useful or overkill?

Reply:
Agree. This is useful in understanding the test better. Added.